diff -Naur hybrid-v35_64-nodebug-pcoem-6_30_223_271.orig/src/include/linuxver.h hybrid-v35_64-nodebug-pcoem-6_30_223_271/src/include/linuxver.h
--- hybrid-v35_64-nodebug-pcoem-6_30_223_271.orig/src/include/linuxver.h	2024-11-26 10:43:08.436733364 +0200
+++ hybrid-v35_64-nodebug-pcoem-6_30_223_271/src/include/linuxver.h		2024-11-26 11:00:51.319204611 +0200
@@ -147,7 +147,7 @@
 #include <linux/sched.h>
 #endif
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 29)
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 29)) && (LINUX_VERSION_CODE <= KERNEL_VERSION(6, 12, 0))
 #include <net/lib80211.h>
 #endif
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 29)
 
